sdk/identity/Azure.Identity/src/CliCredentialClient.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Azure.Identity
+{
+    internal class CliCredentialClient
+    {
+        public virtual (string, int) CreateProcess(string extendCommand)
+        {
+            string fileName = string.Empty;
+            string argument = string.Empty;
+
+            Process proc = new Process();
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                fileName = "cmd";
+                argument = $"/c \"{extendCommand}\"";
+            }
+            else
+            {
+                fileName = "/bin/sh";
```

Comment:
On *nix, is it not possible to just invoke `az` directly?  The script itself is marked executable and has the correct shebang so I believe that we should be able to invoke it directly?  Are we unable to do this because we don't know the full path to the executable?

Reply:
The command can’t execute if there don’t have the full path on *nix.

Reply:
Gotcha.

Reply:
Gotcha